pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -194,11 +194,6 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		gc.workerLock.Lock()
 		defer gc.workerLock.Unlock()
 
-		// Resetting the REST mapper will also invalidate the underlying discovery
-		// client. This is a leaky abstraction and assumes behavior about the REST
-		// mapper, but we'll deal with it for now.
-		gc.restMapper.Reset()
```

Comment:
doesn't the garbage collector also start a gofunc doing this. You'll need to eliminate it.

Reply:
move it here https://github.com/kubernetes/kubernetes/pull/62990/files#diff-2aa9cfe6a01016c1679752b0fcfc9244R404

Reply:
since it is public for controllers

Reply:
> since it is public for controllers

Sorry, I was thinking about the side effect now.  You need to invalidate the discovery client results here because they coupled them :(.